src/daemon/event.c

```diff
@@ -476,11 +476,15 @@ static void
 levent_update_and_send(evutil_socket_t fd, short what, void *arg)
 {
 	struct lldpd *cfg = arg;
-	struct timeval tv = { cfg->g_config.c_tx_interval, 0 };
+	struct timeval tv;
+	long interval_ms = cfg->g_config.c_tx_interval;
+
 	(void)fd; (void)what;
 	lldpd_loop(cfg);
 	if (cfg->g_iface_event != NULL)
-		tv.tv_sec *= 20;
+		interval_ms *= 20;
+	tv.tv_sec = interval_ms / 1000;
+	tv.tv_usec = (interval_ms % 1000) * 1000;
```

Comment:
This is a safeguard to refresh information about all network interfaces at regular interval, in case there was something wrong with the event-based refresh. Maybe you can add a minimal value of 1 minute here.

Reply:
Minimal value of 30 seconds even if we don't have event-based refresh (so outside the `if`). Most people have the event-based refresh, so lowering the pace for the others should be pretty invisible.

Reply:
Do you mean something like this ?

```
@@ -483,6 +483,8 @@ levent_update_and_send(evutil_socket_t fd, short what, void *arg)
        lldpd_loop(cfg);
        if (cfg->g_iface_event != NULL)
                interval_ms *= 20;
+       if (interval_ms < 30000)
+               interval_ms = 30000;
        tv.tv_sec = interval_ms / 1000;
        tv.tv_usec = (interval_ms % 1000) * 1000;
        event_add(cfg->g_main_loop, &tv);

```

Reply:
Yes!

Reply:
I just pushed this change, I quoted you in the commit message, are you OK with this ?